src/api/app/helpers/webui/staging/project_helper.rb

```diff
@@ -0,0 +1,9 @@
+module Webui::Staging::ProjectHelper
+  def icon_for_checks(checks, missing_checks)
+    return 'fa-eye text-info' if missing_checks.present?
+    return 'fa-check-circle text-primary' if checks.blank?
+    return 'fa-eye text-info' if checks.any? { |check| check.state == 'pending' }
```

Comment:
there is a `pending?` function in check, which you can use

Reply:
I would even try to run this on SQL level / via scopes and relations. But I am fine with doing this later, and in another PR.

Reply:
In another PR then :smiley: